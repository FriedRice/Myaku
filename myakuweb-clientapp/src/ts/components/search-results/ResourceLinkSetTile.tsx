/** @module Resource link set tile component */

import React from 'react';
import { ResourceLinkSet } from 'ts/types/types';
import Tile from 'ts/components/generic/Tile';

interface ResourceLinkSetTileProps {
    query: string;
    linkSet: ResourceLinkSet;
}
type Props = ResourceLinkSetTileProps;


const ResourceLinkSetTile: React.FC<Props> = function(props) {
    var resourceLinkLis: React.ReactElement[] = [];
    for (const resourceLink of props.linkSet.resourceLinks) {
        resourceLinkLis.push(
            <li key={resourceLink.resourceName}>
                <a href={resourceLink.link}>
                    {`Search ${resourceLink.resourceName} for `}
                    <span className='japanese-text' lang='ja'>
                        {props.query}
                    </span>
                </a>
            </li>
        );
    }

    return (
        <Tile tileClasses='aside-tile resource-links-tile'>
            <h5>
                {props.linkSet.setName}
            </h5>
            <ul className='resource-links-list'>
                {resourceLinkLis}
            </ul>
        </Tile>
    );
};

export default ResourceLinkSetTile;
