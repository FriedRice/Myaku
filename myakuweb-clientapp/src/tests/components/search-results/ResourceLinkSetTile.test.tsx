/**
 * Tests for the ResourceLinkSetTile component.
 * @module tests/components/search-results/ResourceLinkSetTile.test
 */

import React from 'react';
import ResourceLinkSetTile from
    'ts/components/search-results/ResourceLinkSetTile';
import { shallow } from 'enzyme';

import {
    SEARCH_RESOURCES_NO,
    SEARCH_RESOURCES_OB,
} from 'tests/testData';

describe('<ResourceLinkSetTile />', function() {
    it('renders correctly with resources set', function() {
        const wrapper = shallow(
            <ResourceLinkSetTile
                query={SEARCH_RESOURCES_NO.query}
                linkSet={SEARCH_RESOURCES_NO.resourceLinkSets[0]}
            />
        );
        expect(wrapper).toMatchSnapshot();

        wrapper.setProps({
            query: SEARCH_RESOURCES_OB.query,
            linkSet: SEARCH_RESOURCES_OB.resourceLinkSets[0],
        });
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with no resources set', function() {
        const wrapper = shallow(
            <ResourceLinkSetTile
                query={null}
                linkSet={null}
            />
        );
        expect(wrapper).toMatchSnapshot();

        wrapper.setProps({
            query: SEARCH_RESOURCES_NO.query,
            linkSet: SEARCH_RESOURCES_NO.resourceLinkSets[0],
        });
        expect(wrapper).toMatchSnapshot();
    });
});
