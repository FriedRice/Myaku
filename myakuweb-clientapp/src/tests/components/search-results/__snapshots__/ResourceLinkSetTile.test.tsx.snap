// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResourceLinkSetTile /> renders correctly with no resources set 1`] = `
<Tile
  loadingHeight="6.625rem"
  tileClasses="aside-tile resource-links-tile"
/>
`;

exports[`<ResourceLinkSetTile /> renders correctly with no resources set 2`] = `
<Tile
  tileClasses="aside-tile resource-links-tile"
>
  <h5>
    Jpn-Eng Dictionaries
  </h5>
  <ul
    className="resource-links-list"
  >
    <li
      key="Jisho.org"
    >
      <a
        href="https://jisho.org/search/%E3%81%AE"
      >
        Search Jisho.org for 
        <span
          lang="ja"
        >
          の
        </span>
      </a>
    </li>
    <li
      key="Weblio EJJE"
    >
      <a
        href="https://ejje.weblio.jp/content/%E3%81%AE"
      >
        Search Weblio EJJE for 
        <span
          lang="ja"
        >
          の
        </span>
      </a>
    </li>
  </ul>
</Tile>
`;

exports[`<ResourceLinkSetTile /> renders correctly with resources set 1`] = `
<Tile
  tileClasses="aside-tile resource-links-tile"
>
  <h5>
    Jpn-Eng Dictionaries
  </h5>
  <ul
    className="resource-links-list"
  >
    <li
      key="Jisho.org"
    >
      <a
        href="https://jisho.org/search/%E3%81%AE"
      >
        Search Jisho.org for 
        <span
          lang="ja"
        >
          の
        </span>
      </a>
    </li>
    <li
      key="Weblio EJJE"
    >
      <a
        href="https://ejje.weblio.jp/content/%E3%81%AE"
      >
        Search Weblio EJJE for 
        <span
          lang="ja"
        >
          の
        </span>
      </a>
    </li>
  </ul>
</Tile>
`;

exports[`<ResourceLinkSetTile /> renders correctly with resources set 2`] = `
<Tile
  tileClasses="aside-tile resource-links-tile"
>
  <h5>
    Jpn-Eng Dictionaries
  </h5>
  <ul
    className="resource-links-list"
  >
    <li
      key="Jisho.org"
    >
      <a
        href="https://jisho.org/search/OB"
      >
        Search Jisho.org for 
        <span
          lang="ja"
        >
          OB
        </span>
      </a>
    </li>
    <li
      key="Weblio EJJE"
    >
      <a
        href="https://ejje.weblio.jp/content/OB"
      >
        Search Weblio EJJE for 
        <span
          lang="ja"
        >
          OB
        </span>
      </a>
    </li>
  </ul>
</Tile>
`;
